### PR TITLE
Using bench flow custom format

### LINKED ISSUE #12: Use the custom BenchFlow format to store the data
[Depends on #6] **transform** the data to a custom format used by BenchFlow ([Custom format reference](https://dl.dropboxusercontent.com/u/7530328/github/2015_BenchFlow_Msc_Marco_Argenti.pdf). This reference uses a MySQL database, we should port the format to Cassandra, by taking care of saving the data related to a benchmark_id in the same place. Some hints can be obtained from: http://blog.rackspace.com/cassandra-by-example/)

This issue contains the only point that was not completed in #1

### PR DESCRIPTION
**Pull request by Cerfoglg**
Fixes #12

### CODE REVIEW

File: conf.json [hunk 1]
```diff
@@ -1,10 +1,10 @@
-{ settings: 
+{ "settings": 
 	[
 		{
 		  "src_table": "info",
 		  "dest_table": "info",
-		  "column_mapping": {"ID_": "id", "DURATION_": "start_time"},
-		  "column_transformation": {"DURATION_": "secondsToMilliseconds"}
+		  "column_mapping": {"ID_": "id", "DURATION_": "duration"},
+		  "column_transformation": {}
```
Comment (VincenzoFerme) [inline conf.json]: Put back the example with the column_transformation. It is ok to keep it, even if commented out.

File: .gitignore [hunk 2]
```diff
@@ -1,3 +1,7 @@
 .idea/
 pyspark/
 pyspark_cassandra/
+
+*.pyc
+
+__pycache__/dataTransformations.cpython-35.pyc
```
Comment (VincenzoFerme) [inline .gitignore]: Just ignore the **pycache** folder

File: mysqlTransformer.py [hunk 3]
```diff
@@ -10,11 +10,22 @@
...
 
+# Takes arguments: Spark master, Cassandra host, Minio host, path of the file
+sparkMaster = sys.argv[1]
+cassandraHost = sys.argv[2]
+minioHost = sys.argv[3]
+filePath = sys.argv[4]
+cassandraKeyspace = "benchflow"
+
+# Gets the benchmark ID from the minio file path
+benchmarkID = filePath.split("/")[2]
```
Comment (VincenzoFerme) [inline mysqlTransformer.py]: Why from the path? We can pass it when lunching the script.

File: mysqlTransformer.py [hunk 4]
```diff
@@ -23,15 +34,9 @@ def createDic(a):
         else:
             tf = getattr(Transformations, t)
             d[col] = tf(a[indexes[i]])
+    d["trialid"] = benchmarkID
```
Comment (VincenzoFerme) [inline mysqlTransformer.py:36]: Why the benchmarkID is assigned to the trialid?

File: mysqlTransformer.py [hunk 5]
```diff
@@ -23,15 +34,9 @@ def createDic(a):
...
-# Takes arguments: Spark master, Cassandra host, Minio host, path of the file
-sparkMaster = sys.argv[1]
-cassandraHost = sys.argv[2]
-minioHost = sys.argv[3]
-filePath = sys.argv[4]
-cassandraKeyspace = "test"
-
 # Set configuration for spark context
 conf = SparkConf() \
     .setAppName("PySpark Cassandra Test") \
```
Comment (VincenzoFerme) [inline mysqlTransformer.py]: Use the actual name of the data-transformer: MySQL Transformer

File: mysqlTransformer.py [hunk 6]
```diff
@@ -44,11 +49,11 @@ def createDic(a):
 confPath = SparkFiles.get("conf.json")
 with open(confPath) as f:
     maps = json.load(f)
-    mappings = mappings["settings"]
+    mappings = maps["settings"]
     
 for conf in mappings:
     # Retrieves the file from Minio and parallelize it for Spark
-    res = urllib.request.urlopen("http://"+minioHost+"/"+filePath+"_"+conf["src_table"]+".gz")
+    res = urllib.request.urlopen("http://"+minioHost+":9000/"+filePath+"/"+conf["src_table"]+"csv.gz")
```
Comment (VincenzoFerme) [inline mysqlTransformer.py]: Also the port (9000) should be a configuration variable.

File: statsTransformer.py [hunk 7]
```diff
@@ -14,11 +15,16 @@
 cassandraHost = sys.argv[2]
 minioHost = sys.argv[3]
 filePath = sys.argv[4]
-cassandraKeyspace = "test"
+cassandraKeyspace = "benchflow"
+table = "environmentdata"
+
+# Gets the benchmark ID from the minio file path
+benchmarkID = filePath.split("/")[2]
```
Comment (VincenzoFerme) [inline statsTransformer.py]: Why from the path? We can pass it when lunching the script.

File: statsTransformer.py [hunk 8]
```diff
@@ -34,10 +40,17 @@
...
-    d["usage"] = ob["cpu_stats"]["cpu_usage"]["total_usage"]
+    d["id"] = uuid.uuid1()
+    d["environmentid"] = "something"
+    d["time"] = ob["read"]
+    d["cputotal"] = int(ob["cpu_stats"]["cpu_usage"]["total_usage"])
+    d["memoryused"] = int(ob["memory_stats"]["usage"])
+    d["memorytot"] = int(ob["memory_stats"]["max_usage"])
+    d["networkin"] = int(ob["network"]["rx_bytes"])
+    d["networkout"] = int(ob["network"]["tx_bytes"])
+    d["trialid"] = benchmarkID
```
Comment (VincenzoFerme) [inline statsTransformer.py]: Why the benchmarkID is assigned to the trialid?

**Comment by Cerfoglg** [comment]
Updated with some changes, including using data types for mysql transformer.

File: conf.json [hunk 9]
```diff
@@ -1,9 +1,9 @@
-{ settings: 
+{ "settings": 
 	[
 		{
-		  "src_table": "info",
-		  "dest_table": "info",
-		  "column_mapping": {"ID_": "id", "DURATION_": "start_time"},
+		  "src_table": "Camunda_dump_example_csv",
```
Comment (VincenzoFerme) [inline conf.json]: Here it should be possible to specify the name of the table. Am I correct? If yes, just use an example table name without dump or csv in the name.

Comment (Cerfoglg) [inline conf.json]: @VincenzoFerme Yeah, I was using the dump you gave me as an example to test, but it's meant to be the name of a table.

Comment (simonedavico) [inline conf.json]: Just a question from an outsider: why aren't you using camelCase for JSON properties?

See also [Google JSON style guide](https://google.github.io/styleguide/jsoncstyleguide.xml).

Of course if there is a good reason, ignore this comment.

Comment (VincenzoFerme) [inline conf.json]: @simonedavico good catch. @Cerfoglg why?

Comment (VincenzoFerme) [inline conf.json]: @Cerfoglg @simonedavico I moved the discussion on the following issue https://github.com/benchflow/benchflow/issues/2. @Cerfoglg please explain there the why of your choice.

**Comment by Cerfoglg** [comment]
@VincenzoFerme Updated again to use the updated cassandra design. Note that network values are omitted for now, as that is an issue that needs some extra work.

**Pull request merged**